### PR TITLE
Feeder

### PR DESCRIPTION
**Pull request by veronicawang1**

### CODE REVIEW

File: src/main/kotlin/com/team4099/robot2022/subsystems/Feeder.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  val topBeamBroken: Boolean
+    get() = !topBeamDIO.get()
+
+  val bottomBeamBroken: Boolean
+    get() = !bottomBeamDIO.get()
+
+  var feederState = Constants.Feeder.FeederState.NEUTRAL
+    set(value) {
+      field = value
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/subsystems/Feeder.kt]: could we change the name of value

File: src/main/kotlin/com/team4099/robot2022/commands/feeder/FeederCommand.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import com.team4099.robot2022.subsystems.Feeder
+import edu.wpi.first.wpilibj2.command.CommandBase
+
+class FeederCommand(var dir: Constants.Feeder.FeederState) : CommandBase() {
+  init {
+    addRequirements(Feeder)
+  }
+
+  override fun initialize () {
+    Feeder.feederState = dir
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/commands/feeder/FeederCommand.kt]: why is the state named "dir"

File: src/main/kotlin/com/team4099/robot2022/subsystems/Feeder.kt (hunk 1 above)
Comment (Zenza235) [reply src/main/kotlin/com/team4099/robot2022/subsystems/Feeder.kt]: changed it to "state" if that works

File: src/main/kotlin/com/team4099/robot2022/config/Constants.kt [hunk 3]
```diff
@@ -114,4 +114,30 @@ object Constants {
     const val THROTTLE_DEADBAND = 0.05
     const val TURN_DEADBAND = 0.05
   }
+
+  object Feeder {
+
+    //temp values
+    const val FEEDER_POWER = 1.0
+    const val FLOOR_MOTOR_ID = 1
+    const val VERTICAL_MOTOR_ID = 1
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/config/Constants.kt]: https://docs.google.com/spreadsheets/d/1xyfAiwj5n5N3wEYY8fCAL-Wqd785fyaLiyBSGw-0LKE/edit#gid=52059920
update motor id

File: src/main/kotlin/com/team4099/robot2022/subsystems/Feeder.kt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    // vertical motor values
+    Logger.addSource(Constants.Feeder.TAB, "Feeder Vertical Motor Power") { verticalMotor.motorOutputPercent }
+    Logger.addSource(Constants.Feeder.TAB, "Feeder Vertical Motor Supply Current") { verticalMotor.supplyCurrent }
+    Logger.addSource(Constants.Feeder.TAB, "Feeder Vertical Motor Voltage") { verticalMotor.busVoltage }
+    // beam broken values
+    Logger.addSource(Constants.Feeder.TAB, "Feeder Top Beam DIO Broken") { topBeamBroken }
+    Logger.addSource(Constants.Feeder.TAB, "Feeder Bottom Beam DIO Broken") { bottomBeamBroken }
+    // beam time values
+    Logger.addSource(Constants.Feeder.TAB, "Feeder Beam Broken forwards time") { Constants.Feeder.BEAM_BREAK_BROKEN_TIME }
+    Logger.addSource(Constants.Feeder.TAB, "Feeder Beam Broken backwards time") { Constants.Feeder.BEAM_BREAK_BACKWARDS_TIME }
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/subsystems/Feeder.kt]: in my opinion we dont need to log these values as they are from constants.kt and already know the current time without viewing the logs

File: src/main/kotlin/com/team4099/robot2022/config/Constants.kt [hunk 5]
```diff
@@ -114,4 +114,30 @@ object Constants {
...
+
+    //feeder_power temp
+    const val FEEDER_POWER = 1.0
+    const val FLOOR_MOTOR_ID = 61
+    const val VERTICAL_MOTOR_ID = 62
+    const val TAB = "Feeder"
+
+    //temp values
+    const val TOP_DIO_PIN = 1
+    const val BOTTOM_DIO_PIN = 1
```
Comment (eckshn) [inline src/main/kotlin/com/team4099/robot2022/config/Constants.kt]: we can set one to 0 and the other to 1 as we won't be using any other DIOs

Comment (eckshn) [reply src/main/kotlin/com/team4099/robot2022/config/Constants.kt]: once done with this, the branch looks fine to me so in terminal execute the following command:
./gradlew :spotlessApply 
this allows it to build properly

**Pull request merged**